DOCSP-33246 Convert Views Content (#63)
mongodb/docs-relational-migrator: source/code-generation/query-converter/convert-views.txt

=============

.. contents:: On this page
  :local:
  :backlinks: none
  :depth: 1
  :class: singlecol

You can convert SQL views to MongoDB code with the query
converter. The query converter considers the mapping rules and schema
transformations defined in your project when converting your SQL code.

About this Task
---------------

.. include:: /includes/fact-query-converter-generic.rst

Before you Begin
----------------

.. include:: /includes/fact-query-converter-disclaimer.rst

Steps
-----
.. procedure::

>   :style: normal

Reviewer: Vale raised a false positive for this line. We created https://jira.mongodb.org/browse/DOCSP-34343 to investigate.

  .. step:: Navigate to the query converter pane

     From the :guilabel:`Code Generation` tab, click the
     :guilabel:`Query Converter` pane.

  .. step:: Open the query converter view

     - If it is your first time using the query converter in your
       project, click :guilabel:`Import From Database`.

     - If your project already has converted SQL code, click the
       :guilabel:`Manage Database Objects` button on the left-hand
       pane.

  .. step:: Connect to your relational database

     a. Specify the connection details to your source database. 

        For details, see :ref:`rm-relational-database-connection-strings`.

     #. Click :guilabel:`Connect`.
 
  .. step:: Select stored views

     a. On the :guilabel:`Import Database Objects` modal, click
        :guilabel:`Database`.
     #. Click the :icon-fa5:`chevron-right` icon next to your schema.
     #. Click the :icon-fa5:`chevron-right` icon next to
        :guilabel:`Views`.
     #. You can toggle views for conversion by clicking
        the :icon-fa5:`check-square` icon next to a view's

>         name. All views are selected for conversion by default.

Reviewer: Can we make this step more active voice?

To select the views to convert, click the (...). By default, all views are selected.

     #. Click :guilabel:`Save`.

  .. step:: Convert and test code

     a. Click a view's name from the left-hand pane under
        :guilabel:`Views`.
        

>         The SQL code is loaded into the :guilabel:`Imported View`

Reviewer: ```suggestion
        The SQL code is displayed in the :guilabel:`Imported View`
```

        pane.
     #. Click the :guilabel:`Convert` button. Wait for the query 
        converter to convert your code. 
        
        The converted MongoDB code 

>         displays in the :guilabel:`Converted MongoDB Query` pane.

Reviewer: ```suggestion
        is displayed in the :guilabel:`Converted MongoDB Query` pane.
```

        
     #. Click the :icon-fa5:`copy` icon to copy the MongoDB code
        to your clipboard.
     #. Test and validate the generated MongoDB code.

     .. tip::

        .. include:: /includes/fact-query-converter-filter.rst

Examples
--------

> The query converter can convert your SQL views to MongoDB code. 

Reviewer: I don't think we need this initial sentence.

> The examples below show PostGreSQL views from the

Reviewer: ```suggestion
The examples below show PostgreSQL views from the
```

:ref:`example Northwind database<create-project-sample-schema>` 
converted to MongoDB code.

> The following example shows a ``inner join`` view converted to MongoDB:

Reviewer: ```suggestion
The following example shows an ``inner join`` view converted to MongoDB:
```

> The following example shows a ``inner join`` view converted to MongoDB:

Reviewer: [question]

Does "inner join" need to be monospaced?

> .. tabs::

Reviewer: I'm not sure the tabs are particularly helpful here. I think it's helpful to be able to see the input and output at once. Tabs would make sense if we were maybe showing different types of relational databases (in addition to Postgres), but I think in this case we might want to show both code-blocks together.
